apps/channels/tasks.py

```diff
@@ -56,6 +57,21 @@ def handle_twilio_message(self, message_data: str):
     message_handler.new_user_message(message)
 
 
+@shared_task(bind=True, base=TaskbadgerTask)
+def handle_sureadhere_message(self, client_id: str, message_data: dict):
+    message = SureAdhereMessage.parse(message_data)
+    experiment_channel = ExperimentChannel.objects.filter(
+        extra_data__client_id=client_id,
+        platform=ChannelPlatform.IN_APP,
+        messaging_provider__type=MessagingProviderType.sureadhere,
```

Comment:
Surely it's not possible to create a channel with this platform and a different messaging provider type?